### DISCUSSION TITLE
Messaging MVP

### DISCUSSION THREAD

**Discussion by joemasilotti** [joemasilotti/railsdevs.com#72] [General]
The ability to directly email anyone on the site needs to be rethought. This is opening up the opportunity for someone to spam and/or harass folks by simply signing up for an account. Immediately following this being posted, the Message button will be removed from the site.

The Message button will not be added back in until the following criteria is met:

- [x] A developer's email address is hidden to everyone using the site
- [x] The site has a clear description and purpose (see #71)
- [x] The site documents the creators and a contact email
- [x] In-app messaging is built out so someone can contact another developer _via the site_
- [ ] Moderation and flagging for abuse, harassment, and spam for each message and user is implemented
- [ ] The site clearly states it's code of conduct

**Comment by jacobdaddario** [comment]
I'm gonna be home this weekend. I'm super excited to wrap things up on those ActiveStorage test PRs and pluck away at some of these.

**Comment by joemasilotti** [comment]
[About page](https://www.railsdevs.io/about) is now live!

**Comment by joemasilotti** [comment]
I've kicked off an official project for [Messaging](https://github.com/joemasilotti/railsdevs.com/projects/3). There you can find all the (known) to-dos that are needed to get an MVP live. I've also outlined the first five issues with more a more detailed implementation checklist.

**Reply by joemasilotti** [reply]
I opened the first PR, #125. This creates a new flow after signing up to self-identify as a business or developer and redirect to the appropriate form.

**Reply by joemasilotti** [reply]
#125 has been merged in and is behind a feature flag.

You can create a business everywhere but in production. If you create one on your machine, let me know if you have feedback.

**Comment by joemasilotti** [comment]
I submitted PR #121! I'm going to sleep on it before doing a self-review. But if anyone takes a look before me feel free to let me know what you think.

**Comment by joemasilotti** [comment]
I've also opened #132 which quickly outlines a potential problem with the modeling. How can we determine the "intent" of the user - are they acting as a developer or as a business? Would love feedback if anyone has ideas.

**Comment by joemasilotti** [comment]
I've decided that splitting out #120, #121, and #122 into separate PRs will end up being more work than doing them all at once. This is because the modeling required to get #120 done has to be reworked for #121. And for #122 to work the routes need to change (again).

Looking forward, even though it's a huge feature, building both sides of the marketplace at the same time might make more sense. For example, if/when we add notifications or email support we should build both sides at the same time.

So, #132 is ready for review! I've asked @jacobdaddario and @Tonksthebear to take a peek (but no pressure, of course) since they expressed interest in the features before. I'm hoping to get this code live this week so we can move on to moderation and the likes.

In the meantime, I'll be creating new issues for moderation and taking a second pass at the homepage messaging.

**Comment by joemasilotti** [comment]
The Messaging MVP project is officially closed! I haven't flipped the `Feature` toggle yet, but all the code is live in production.

There's a lot of follow up work to be done, but it all should technically work now. I'm going to document next steps soon (mostly related to Payments and Notifications).
